packages/core/src/components/axes/axis.ts

```diff
@@ -70,13 +70,44 @@ export class Axis extends Component {
 				break;
 		}
 
+		// Add axis into the parent
+		const container = DOMUtils.appendOrSelect(svg, `g.axis.${axisPosition}`);
+		const axisRefExists = !container.select(`g.ticks`).empty();
+		let axisRef = DOMUtils.appendOrSelect(container, `g.ticks`);
+		if (!axisRefExists) {
+			axisRef.attr("role", `${Roles.GRAPHICS_OBJECT} ${Roles.GROUP}`);
+		}
+
+		// We draw the invisible axis because of the async nature of d3 transitions
+		// To be able to tell the final width & height of the axis when initiaing the transition
+		// The invisible axis is updated instantly and without a transition
+		const invisibleAxisRef = DOMUtils.appendOrSelect(container, `g.ticks.invisible`)
+			.style("opacity", "0")
+			.attr("aria-hidden", true);
+
+		// Get font height
+		const fakeTick = DOMUtils.appendOrSelect(invisibleAxisRef, `g.tick`);
+		const fakeTickText = DOMUtils.appendOrSelect(fakeTick, `text`).text("0");
+		const tickHeight = fakeTickText.node().getBBox().height;
```

Comment:
`getBBox()` doesn't always work in some browser versions. Use `DOMUtils.getSVGElementSize()` please, search for usage examples in the code

Reply:
👍